posthog/queries/base.py

```diff
@@ -240,9 +240,16 @@ def empty_or_null_with_value_q(
     else:
         parsed_value = None
         if operator in ("gt", "gte", "lt", "lte"):
+            if isinstance(value, list):
+                # If the value is a list for these operators,
+                # we should not return any results, as we can't compare a list to a single value
+                # TODO: should we try and parse each value in the list and return results based on that?
+                return Q(pk__isnull=True)
```

Comment:
I made the executive decision to just null the filter condition if there's an invalid comparison (i.e., just fail to evaluate the flag to true if the user made an invalid filter).  It's a bit of a "throw our hands up and ensure that the filter logic always serves us correct data" situation.

That said, another approach here would be to attempt to parse the invalid filter condition and try and do the right thing (in this case, we could just extract the value from the array and then generate the filter).  Open to feedback on that; I feel like it may solve a bunch of user issues but potentially introduce new buggy behavior that users don't expect.  It might be the better idea, though.

Reply:
Oh yeah i ran into that case when going through my tests; I think I can do the same thing there.  Will look, and thanks for the context!

Reply:
Thanks for the feedback!  I made your suggested changes and wrote some [tests](https://github.com/PostHog/posthog/pull/23297/files#diff-7d9c4d6fc0009dff4f4c3a7b27b40f6e7d1d3eb0cc634a8f0bd1621028b19aa0L89)